Process for inactive contributions

jasonblais opened the pull request.

jasonblais commented:
@hanzei Would love to hear your thoughts on the proposed process either here or in our DM channel.

hanzei commented:
Responded via DM

jasonblais commented:
@hanzei I made a couple of updates following our discussion. Let me know your thoughts?

Also adding Joram as a second reviewer.

jasonblais commented:
> We should probably make some sort of process announcement the rest of the core committers are aware

Oh right, Hanzei and I talked about bringing it up in a dev meeting. @jwilander would you be open to reviewing in a dev meeting? Or I can bring up in platform meeting if that's a better avenue.

hanzei commented on site/content/contribute/getting-started/inactive-contribution.md:
```diff
@@ -0,0 +1,26 @@
+subsection: Getting Started
+---
+
+# Inactive Contributions
```
Because of https://github.com/mattermost/mattermost-developer-documentation/pull/197 this line should be removed.

jwilander commented:
Bringing it up in both platform and the dev meetings make sense to me

jasonblais commented:
Thanks, queued for platform, @jwilander please help queue for dev meeting if beneficial?

lieut-data commented on site/content/contribute/getting-started/inactive-contribution.md:
```diff
@@ -0,0 +1,24 @@
+ - If action is required from reviewers, Community Coordinator asks reviewers to share feedback or help answer questions, and adds `stale` label to the contribution. The Coordinator will follow up with reviewers until a response is received.
+
+2. After another 14 days of inactivity, a contribution becomes rotten.
+ - Community Coordinator asks the submitter if the team can help with questions. They acknowledge that after another 30 days of inactivity the contribution will be closed. They also add a `rotten` label to the contribution.
```
Is `rotten` is an industry standard term? It has a decidedly negative connotation to me that `stale` or `inactive` don't invoke.

jasonblais replied:
The labels were inspired by Kubernetes community: https://github.com/kubernetes/kubernetes/labels?utf8=%E2%9C%93&q=lifecycle

I'm definitely open to another term though 👍

lieut-data replied:
Honestly, I'm 0/5 on this -- if it's a standard term, I think there's value in that and I can just interpret the term differently :)

hanzei replied:
:+1: on the negative connotation.
I'm not an native speaker and con't think of a good alternative. Maybe `orphaned` or something similar to this?

jasonblais replied:
Cool - I've heard users referring to "dead" channels/teams as `dormant` so that may be another option. Though `orphaned` is likely more familiar with non-native speakers.

So we can go with `stale` > `inactive` > `orphaned` - thoughts? (I feel that `orphaned` implies more strongly that the PR won't be worked on anymore, whereas `inactive` has the possibility of being re-activated).

lieut-data replied:
So I wonder if we could just make do with `stale` and `inactive`, closing the issue on the transition to `inactive`. It's not like that's the end: the ticket can always be reopened, but that's already 44 days of inactivity which seems long enough IMO.

hanzei replied:
I'm not sure where do you get the 44 days from. The process is, I think,
14ht day: Add `stale`
28th day: Remove `stale` and add `inactive`
58th day: Remove `inactive`, add `orphaned` and close PR.

The main advantage of having a label just for PR's that got closed do to inactivity is, that you can search for them.

lieut-data replied:
@hanzei, so I was wondering if it made sense to just have two lifecycle labels here: `stale` after 14 days, and `inactive` (or `orphaned`) after 30 days, after which the issue would be closed. Seems like it might be easier to manage and interpret at a glance, vs. trying to remember which of `stale` and `inactive` "came first".

In the three-lifecycle-label proposal, what's the advantage of the extra label?

jasonblais replied:
The current process reaches out the community member twice (on day 14 and day 28) before closing the PR on day 58.

I'm 0/5 between a 2 or 3 lifecycle-label process - two is simpler, and I'm happy to go with that, as long as we then contact the contributor twice during the `stale` period.

--

On a side note, as we scale, I'm wondering how easy/hard it would be to add capability for [mattermod](https://github.com/mattermost/mattermost-mattermod) to automatically contact contributor and apply these labels after X number of days.

lieut-data replied:
0/5 as well, so long as we number the lifecycles, e.g. `0:stale`, `-1:orphaned`, `-2:inactive` or some such.

Love the `mattermod` idea, and honestly wouldn't be very hard! I'd say we file a ticket as part of this effort and automate it from the beginning.

hanzei replied:
:+1: on the numbering of the lables.
:+1: on the `mattermod` idea.

I like the 3 label process because it makes the workflow clear. Every time we reached out to the contributor, we add a new "level" of label. And reaching out to the contributor twice seams is important in my opinion.

lieut-data replied:
@hanzei, framing it as the number of times we reach out to a contributor really helps clarify the proposal for me. Makes a ton of sense!

hanzei replied:
Did we agree on calling the 2th step `orphaned`?

jasonblais replied:
I was thinking stale > orphaned > inactive. Not that we should follow Kubernetes community 1-1, but their first state is `stale` and last state is `inactive`, so I'd prefer to keep that consistent. That way contributors and users switching between the two repos have the same understanding of the inactive state.

hanzei replied:
After think a bit about this, I would consider going with stale > inactive > orphaned. orphaned seams to me (a non native speaker) as the strongest of all three words. It also describes what happened quite well: The contributor leaved the contribution alone.

I don't think the naming of Kubernetes is so idiomatic that we must follow it. Keep in mind that we also number the steps and its quite easy to pick up the progress.

lieut-data commented on site/content/contribute/getting-started/inactive-contribution.md:
```diff
@@ -8,17 +8,17 @@ subsection: Getting Started
 1. After 14 days of inactivity, a contribution becomes stale.
- - If action is required from submitter, Community Coordinator asks if the team can help clarify previous feedback or provide guidance on next steps, and adds `stale` label to the contribution.
- - If action is required from reviewers, Community Coordinator asks reviewers to share feedback or help answer questions, and adds `stale` label to the contribution. The Coordinator will follow up with reviewers until a response is received.
+ - If action is required from submitter, Community Coordinator asks if the team can help clarify previous feedback or provide guidance on next steps, and adds `lifecycle/0:stale` label to the contribution.
```
0/5, but I originally proposed starting at `0` as a cute way of counting negatively. If we're sticking with positive numbers, perhaps start at `1` for uniformity with our other labels?

jasonblais replied:
+1

jasonblais commented:
@hanzei @lieut-data 

1. Updated labels to `lifecycle/1:stale`, `lifecycle/2:rotten`, `lifecycle/3:inactive`, `lifecycle/frozen` as suggested. I added the `lifecycle` prefix so not interfere our existing labels with `1: PM Review`, `2: Dev Review`, etc.
2. Ticket to add capability for mattermod to carry out this process automatically: https://mattermost.atlassian.net/browse/MM-13534
 - this could be a good help wanted ticket if the team doesn't have time to work on it at the moment

hanzei commented:
FYI: I've made a last minute comment about the naming of the labels https://github.com/mattermost/mattermost-developer-documentation/pull/189#discussion_r242906723

Let's finish the discussion in this PR and, if needed, change it it another PR.

jasonblais commented:
@hanzei I'm okay with changing the order to stale/inactive/orphaned. I'm only 1/5 on the current ordering, so I've submitted a PR to change it https://github.com/mattermost/mattermost-developer-documentation/pull/209

We can always adjust later on if needed
